rct229/rulesets/ashrae9012019/ruleset_functions/get_surface_conditioning_category_dict.py

```diff
@@ -154,13 +154,24 @@ def get_surface_conditioning_category_dict(climate_zone, building):
         # Loop through all the surfaces in the zone
         for surface in find_all("surfaces[*]", zone):
             surface_adjacent_to = surface["adjacent_to"]
-            surface_conditioning_category_dict[surface["id"]] = SCC_DATA_FRAME.at[
-                # row index
-                zcc,
-                # column index
+            adjacency = (
                 zcc_dict[getattr_(surface, "surface", "adjacent_zone")]
                 if surface_adjacent_to == SurfaceAdjacency.INTERIOR
-                else surface_adjacent_to,
-            ]
+                else surface_adjacent_to
+            )
+
+            if zcc in SCC_DATA_FRAME.index and adjacency in SCC_DATA_FRAME.columns:
+                surface_conditioning_category_dict[surface["id"]] = SCC_DATA_FRAME.at[
+                    zcc,  # row index
+                    adjacency,  # column index
+                ]
+            elif adjacency in [SurfaceAdjacency.IDENTICAL, SurfaceAdjacency.UNDEFINED]:
+                surface_conditioning_category_dict[
+                    surface["id"]
+                ] = SurfaceConditioningCategory.UNREGULATED
```

Comment:
Any reason we put this logic to `elif`?
I wonder if it can start at `if` to prevent any future accident modifications on the `SCC_DATA_FRAME`.  Putting the `adjacency in [SurfaceAdjacency.IDENTICAL, SurfaceAdjacency.UNDEFINED]` at first ensures the `UNREGULATED` gets returned.

Reply:
Ok that makes sense to me